src/uu/cp/src/cp.rs

```diff
@@ -1286,10 +1286,13 @@ fn copy_file(
         handle_existing_dest(source, dest, options)?;
     }
 
-    // Fail if dest is a dangling symlink or a symlink this program created previously
+    // Fail if upon dereferencing symlink,
+    // dest a dangling symlink or a symlink this program created previously.
+
     if fs::symlink_metadata(dest)
         .map(|m| m.file_type().is_symlink())
         .unwrap_or(false)
+        && !options.dereference
```

Comment:
maybe swapping the two clauses of the and test should be better to short-circuit the fs call.